File: user-service/src/main/java/rs/raf/user_service/domain/entity/Permission.java
package rs.raf.user_service.domain.entity;

import lombok.*;

import javax.persistence.*;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@RequiredArgsConstructor
public class Permission {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String name;
}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/domain/entities/Permission.kt
package pack.userservicekotlin.domain.entities

import jakarta.persistence.*

@Entity
open class Permission(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null,
    @Column(unique = true, nullable = false)
    open var name: String? = null,
)
